### PR TITLE
feat: unstable_dev fetch allow custom path

### PR DESCRIPTION
**Pull request by cameron-robey**
Allows to specify a path in `unstable_dev`.

```js
const worker = await unstable_dev(
  "script.js"
);
const res = await worker.fetch(req);
```

where `req` can be anything from `RequestInfo`: `string | URL | Request`.

Closes https://github.com/cloudflare/wrangler2/issues/1752

### CODE REVIEW

File: packages/wrangler/src/__tests__/api-dev.test.ts [hunk 1]
```diff
@@ -16,4 +21,192 @@ describe("unstable_dev", () => {
 		}
 		await worker.stop();
 	});
+
+	describe("protocol", () => {
+		it("should use https localProtocol", async () => {
+			const port = 8080;
+			msw.use(
```
Comment (rozenmd) [inline packages/wrangler/src/__tests__/api-dev.test.ts:28]: You don't need to use MSW here btw - this test uses an actual fetch against a running worker

Comment (JacobMGEvans) [reply packages/wrangler/src/__tests__/api-dev.test.ts]: It was some hacky stuff I was trying. Dont worry.

**Comment by cameron-robey** [comment]
Tests failing noted in https://github.com/cloudflare/wrangler2/issues/1842

Also failing on other PRs

**Comment by JacobMGEvans** [comment]
Tests passing locally for Wrangler and current changes.

**Review by JacobMGEvans — APPROVED** [review]

File: .changeset/empty-cobras-agree.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+---
+"wrangler": patch
+---
+
+feat: allow specifying a path with unstable_dev fetch
+
+Allows to specify a path in `unstable_dev`.
```
Comment (rozenmd) [inline .changeset/empty-cobras-agree.md:7]:
```suggestion
```

File: .changeset/empty-cobras-agree.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+```
+const worker = await unstable_dev(
+  "script.js", {}, {}
```
Comment (rozenmd) [inline .changeset/empty-cobras-agree.md:11]:
```suggestion
  "script.js",
```
(if they're empty, shouldn't need them)

File: packages/wrangler/src/api/dev.ts [hunk 4]
```diff
@@ -1,9 +1,9 @@
-import { fetch } from "undici";
+import { fetch, Request } from "undici";
```
Comment (rozenmd) [inline packages/wrangler/src/api/dev.ts:1]: dumb question, should this be imported as a type?

**Review by rozenmd — COMMENTED** [review]
Did a quick look through, will re-review tomorrow morning when my eyes aren't fried 😂

File: packages/wrangler/src/api/dev.ts (hunk 4 above)
Comment (cameron-robey) [reply packages/wrangler/src/api/dev.ts:1]: Indeed it should, not a dumb question at all

File: packages/wrangler/src/__tests__/api-dev.test.ts [hunk 5]
```diff
@@ -16,4 +20,182 @@ describe("unstable_dev", () => {
 		}
 		await worker.stop();
 	});
+
+	describe("protocol", () => {
+		it("should use parse to give https url with localProtocol", async () => {
+			const [input, _] = parseRequestInput(
```
Comment (rozenmd) [inline packages/wrangler/src/__tests__/api-dev.test.ts:26]: Can we split out the tests for helper functions like `parseRequestInput` into their own test file?

Just wanted to keep `api-dev.test.ts` for testing `unstable_dev` directly

**Review by rozenmd — CHANGES_REQUESTED** [review]

File: packages/wrangler/src/__tests__/api-dev.test.ts (hunk 5 above)
Comment (cameron-robey) [reply packages/wrangler/src/__tests__/api-dev.test.ts]: https://github.com/cloudflare/wrangler2/pull/1839/commits/63a3fe11f767e9c27fe3dd6bdd2da517d7302ae7

**Review by rozenmd — APPROVED** [review]
LGTM! 🎉

**Pull request merged**